pkg/translate/translate.go

```diff
@@ -36,59 +37,91 @@ type Translation struct {
 
 		"Security.ControlPlaneMtls.Value":    {"global.controlPlaneSecurityEnabled", "", nil},
 		"Security.DataPlaneMtlsStrict.Value": {"global.mtls.enabled", "", nil},
-		"Security.TrustDomain":               {"global.trustDomain", "", nil},
-		"Security.SelfSigned.Value":          {"security.selfSigned", "", nil},
-		"Security.CreateMeshPolicy.Value":    {"security.createMeshPolicy", "", nil},
 	}
-)
-
-// defaultTranslationFunc is the default translation to values. It maps a Go data path into a YAML path.
-func defaultTranslationFunc(m *Translation, root util.Tree, valuesPath string, value interface{}) error {
-	var path []string
 
-	if util.IsEmptyString(value) {
-		dbgPrint("Skip empty string value for path %s", m.k8sPath)
-		return nil
-	}
-	if valuesPath == "" {
-		dbgPrint("Not mapping to values, resources path is %s", m.k8sPath)
-		return nil
+	// ComponentToHelmValuesName is the root component name used in values YAML files in component charts.
+	ComponentToHelmValuesName = map[name.ComponentName]string{
+		name.IstioBaseComponentName:       "global",
+		name.PilotComponentName:           "pilot",
+		name.GalleyComponentName:          "galley",
+		name.SidecarInjectorComponentName: "sidecarInjectorWebhook",
+		name.PolicyComponentName:          "mixer.policy",
+		name.TelemetryComponentName:       "mixer.telemetry",
+		name.CitadelComponentName:         "citadel",
+		name.NodeAgentComponentName:       "nodeAgent",
+		name.CertManagerComponentName:     "certManager",
+		name.IngressComponentName:         "gateways.istio-ingressgateway",
+		name.EgressComponentName:          "gateways.istio-ingressgateway",
 	}
 
-	for _, p := range util.PathFromString(valuesPath) {
-		path = append(path, firstCharToLower(p))
+	FeatureComponentToValues = map[name.FeatureName]map[name.ComponentName]componentValuePaths{
+		name.TrafficManagementFeatureName: {
+			name.PilotComponentName: {
+				enabled:   "pilot.enabled",
+				namespace: "global.istioNamespace",
+			},
+		},
+		name.PolicyFeatureName: {
+			name.PolicyComponentName: {
+				enabled:   "mixer.policy.enabled",
+				namespace: "global.policyNamespace",
+			},
+		},
+		name.TelemetryFeatureName: {
+			name.TelemetryComponentName: {
+				enabled:   "mixer.telemetry.enabled",
+				namespace: "global.telemetryNamespace",
+			},
+		},
+		// TODO: check if these really should be settable.
+		name.SecurityFeatureName: {
+			name.NodeAgentComponentName: {
+				enabled:   "nodeagent.enabled",
+				namespace: "global.istioNamespace",
+			},
+			name.CertManagerComponentName: {
+				enabled:   "certmanager.enabled",
+				namespace: "global.istioNamespace",
+			},
+			name.CitadelComponentName: {
+				enabled:   "security.enabled",
+				namespace: "global.istioNamespace",
+			},
+		},
 	}
-
-	return setYAML(root, path, value)
-}
+)
 
 // ProtoToValues traverses the supplied InstallerSpec and returns a values.yaml translation from it. Mappings defines
```

Comment:
The comments need to be updated as well.

Reply:
Done.